src/platforms/desktop.c

```diff
@@ -69,6 +69,24 @@ void platform_init(void) {
         log_fatal("Error initializing SDL");
     }
 
+    // Set the scaling algorithm to nearest-neighbor for crisp pixel art rendering.
+    // This ensures that pixel art graphics remain sharp when scaled.
+    SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "0");
+
+    // Disable the X11 net WM ping protocol to potentially improve rendering consistency.
+    // This is more relevant for X11 systems but doesn't hurt to include for broader compatibility.
+    SDL_SetHint(SDL_HINT_VIDEO_X11_NET_WM_PING, "0");
+
+    // Force SDL to use OpenGL as the rendering driver.
+    // NOTE: This is somewhat inconsistent with the overall approach of this file (SDL 2D API),
+    // but it helps solve issues with blurry rendering on some systems, particularly high-DPI displays.
+    // Potential implications:
+    // 1. May improve performance and pixel-perfect rendering on some systems;
+    // 2. Could potentially cause compatibility issues on systems without good OpenGL support;
+    // 3. Might have unexpected behavior or performance impacts on certain configurations.
+    // TODO: Consider making this configurable or exploring alternative solutions for blurriness issues
+    SDL_SetHint(SDL_HINT_RENDER_DRIVER, "opengl");

```

Comment:
This platform is not opengl. Since it's only a hint it shouldn't hurt?

Reply:
Though I think on windows it uses DirectX under the hood? I'd prefer SDL to choose what it thinks is best for the non-opengl platform.

Reply:
Please check out my second commit, it should fix the DirectX problem.